src/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs

```diff
@@ -0,0 +1,318 @@
+                        }
+                    }
+                    connectionString = builder.ToString();
+                    _connectionString = connectionString;
+                }
+                return connectionString;
+            }
+            set
+            {
+                DbConnectionOptions constr = new DbConnectionOptions(value, null);
+                string originalValue = ConnectionString;
+                Clear();
+                try
+                {
+                    for (NameValuePair pair = constr.KeyChain; null != pair; pair = pair.Next)
+                    {
+                        if (null != pair.Value)
+                        {
+                            this[pair.Name] = pair.Value;
+                        }
+                        else
+                        {
+                            Remove(pair.Name);
+                        }
+                    }
+                    _connectionString = null;
+                }
+                catch (ArgumentException)
+                { // restore original string
+                    ConnectionString = originalValue;
+                    _connectionString = originalValue;
+                    throw;
+                }
+            }
+        }
+
+        public virtual int Count
+        {
+            get { return CurrentValues.Count; }
+        }
+
+        public bool IsReadOnly
+        {
+            get { return false; }
+        }
+
+        public virtual bool IsFixedSize
+        {
+            get { return false; }
+        }
+        bool System.Collections.ICollection.IsSynchronized
+        {
+            get { return Collection.IsSynchronized; }
+        }
+
+        public virtual ICollection Keys
+        {
+            get
+            {
+                return Dictionary.Keys;
+            }
+        }
+
+
+        object System.Collections.ICollection.SyncRoot
+        {
+            get { return Collection.SyncRoot; }
+        }
+
+        public virtual ICollection Values
+        {
+            get
+            {
+                System.Collections.Generic.ICollection<string> keys = (System.Collections.Generic.ICollection<string>)Keys;
+                System.Collections.Generic.IEnumerator<string> keylist = keys.GetEnumerator();
+                object[] values = new object[keys.Count];
+                for (int i = 0; i < values.Length; ++i)
+                {
+                    keylist.MoveNext();
+                    values[i] = this[keylist.Current];
+                    Debug.Assert(null != values[i], "null value " + keylist.Current);
+                }
+                return new System.Data.Common.ReadOnlyCollection<object>(values);
+            }
+        }
+
+        void System.Collections.IDictionary.Add(object keyword, object value)
+        {
+            Add(ObjectToString(keyword), value);
+        }
+        public void Add(string keyword, object value)
+        {
+            this[keyword] = value;
```

Comment:
This is the desired behavior. The connection string builder needs to have a single value for any connection string key.